tools/mc-java/src/main/java/io/spine/tools/mc/java/codegen/MessagesConfig.java

```diff
@@ -0,0 +1,137 @@
+                           .setProperty(String.class)
+                           .convention(ImmutableList.of());
+        nestedClassFactories = p.getObjects()
+                                .setProperty(String.class)
+                                .convention(ImmutableList.of());
+    }
+
+    void emptyByConvention() {
+        interfaceNames().convention(ImmutableSet.of());
+        methodFactories.convention(ImmutableSet.of());
+        nestedClassFactories.convention(ImmutableSet.of());
+    }
+
+    /**
+     * Specifies a {@link io.spine.tools.protoc.MethodFactory} to generate methods for
+     * the message classes.
+     *
+     * <p>Calling this method multiple times will add provide factories for code generation.
+     *
+     * @param factoryClassName
+     *         the canonical class name of the method factory
+     */
+    public void generateMethodsWith(String factoryClassName) {
+        methodFactories.add(factoryClassName);
+    }
+
+    /**
+     * Specifies a {@link io.spine.tools.protoc.NestedClassFactory} to generate nested classes
+     * inside the message classes.
+     *
+     * <p>Calling this method multiple times will add provide factories for code generation.
+     *
+     * @param factoryClassName the canonical class name of the method factory
+     */
+    public void generateNestedClassesWith(String factoryClassName) {
+        nestedClassFactories.add(factoryClassName);
+    }
+
+    @Override
+    ForMessages toProto() {
+        ForMessages.Builder result = ForMessages.newBuilder()
+                .setPattern(pattern)
+                .addAllAddInterface(interfaces())
+                .addAllGenerateMethods(generateMethods())
+                .addAllGenerateNestedClasses(generateNestedClasses());
+        GenerateFields generateFields = generateFields();
+        if (isNotDefault(generateFields)) {
+            result.setGenerateFields(generateFields);
+        }
+        return result.build();
+    }
+
+    private Set<GenerateMethods> generateMethods() {
+        return methodFactories.get()
+                              .stream()
+                              .map(name -> MethodFactoryName.newBuilder()
+                                      .setClassName(className(name))
+                                      .build())
+                              .map(name -> GenerateMethods.newBuilder()
+                                      .setFactory(name)
+                                      .build())
+                              .collect(toSet());
+    }
+
+    private Set<GenerateNestedClasses> generateNestedClasses() {
+        return nestedClassFactories.get()
+                                   .stream()
+                                   .map(name -> NestedClassFactoryName.newBuilder()
+                                           .setClassName(className(name))
+                                           .build())
+                                   .map(name -> GenerateNestedClasses.newBuilder()
```

Comment:
Please simplify the `name` -> `GenerateNestedClasses` conversion by extracting it to a method.